libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp

```diff
@@ -377,6 +377,9 @@ void NavEKF3_core::setAidingMode()
             meaHgtAtTakeOff = baroDataDelayed.hgt;
             // reset the vertical position state to faster recover from baro errors experienced during touchdown
             stateStruct.position.z = -meaHgtAtTakeOff;
+            // store the current height to be used to keep reporting
+            // the last known position
+            lastKnownPositionD = stateStruct.position.z;
```

Comment:
are  you sure we should save the pos after the reset ?

Reply:
This is mirroring what we're doing to x/y....

Reply:
This is a fallback to a non navigation mode of the the EKF so this reset order is necessary if we are to continue reporting the last known position.